### PR TITLE
feat: deprecate `Deno.close()`

### PR DESCRIPTION
**Pull request by iuioiua**
For removal in Deno v2.

### CODE REVIEW

File: ext/node/polyfills/_fs/_fs_close.ts [hunk 1]
```diff
@@ -11,7 +11,7 @@ export function close(fd: number, callback: CallbackWithError) {
   setTimeout(() => {
     let error = null;
     try {
-      Deno.close(fd);
+      Deno[Deno.internal].core.close(fd);
```
Comment (littledivy) [inline ext/node/polyfills/_fs/_fs_close.ts:14]:
```ts
import { core } from "ext:core/mod.js";

// ...

// TODO: Treat `fd` as real file descriptor. `rid` is an implementation detail and may change.
core.close(fd);
```

File: ext/node/polyfills/_fs/_fs_close.ts [hunk 2]
```diff
@@ -21,5 +21,5 @@ export function close(fd: number, callback: CallbackWithError) {
 
 export function closeSync(fd: number) {
   fd = getValidatedFd(fd);
-  Deno.close(fd);
+  Deno[Deno.internal].core.close(fd);
```
Comment (littledivy) [inline ext/node/polyfills/_fs/_fs_close.ts:24]: Ditto

File: ext/node/polyfills/internal/fs/handle.ts [hunk 3]
```diff
@@ -134,7 +134,7 @@ export class FileHandle extends EventEmitter {
 
   close(): Promise<void> {
     // Note that Deno.close is not async
-    return Promise.resolve(Deno.close(this.fd));
+    return Promise.resolve(Deno[Deno.internal].core.close(this.fd));
```
Comment (littledivy) [inline ext/node/polyfills/internal/fs/handle.ts:137]: Use `core.close` from import

File: ext/node/polyfills/_fs/_fs_close.ts (hunk 1 above)
Comment (iuioiua) [reply ext/node/polyfills/_fs/_fs_close.ts:14]: `ensureFastOps()`?

Comment (littledivy) [reply ext/node/polyfills/_fs/_fs_close.ts]: `core.close` should work fine, it internally calls an op returned by `ensureFastOps()`

**Review by bartlomieju — APPROVED** [review]

**Pull request merged**